runtime/protocol/http-client-engines/http-client-engine-crt/common/src/aws/smithy/kotlin/runtime/http/engine/crt/RequestUtil.kt

```diff
@@ -61,3 +61,11 @@ internal fun HttpRequest.toCrtRequest(callContext: CoroutineContext): aws.sdk.ko
 
     return aws.sdk.kotlin.crt.http.HttpRequest(method.name, url.encodedPath, crtHeaders.build(), bodyStream)
 }
+
+/**
+ * @return whether this HttpRequest is a chunked request.
+ * Specifically, this means return `true` if a request contains a `Transfer-Encoding` header with the value `chunked`,
+ * and the body is either [HttpBody.SourceContent] or [HttpBody.ChannelContent].
+ */
+internal val HttpRequest.isChunked: Boolean get() = (this.body is HttpBody.SourceContent || this.body is HttpBody.ChannelContent) &&
+    this.headers.getAll("Transfer-Encoding")?.contains("chunked") == true
```

Comment:
nit: `headers.contains("Transfer-Encoding", "chunked")` already checks if any header contains the value

nit 2: You can generally leave off qualifying `this` on extension functions most of the time.